agent/engine/execcmd/manager_start_linux.go

```diff
@@ -42,7 +44,7 @@ func (m *manager) RestartAgentIfStopped(ctx context.Context, client dockerapi.Do
 		return NotRestarted, nil
 	}
 	execMD := container.GetExecCommandAgentMetadata()
-	if execMD == nil { // This means m.Start has never been invoked
+	if execMD == nil || execMD.StartedAt == nil { // This means m.Start has never been invoked
```

Comment:
can we encapsulate this condition in a function of `*manager`?

```golang
func (m *manger) isAgentStarted(execMD *container.ExecCommandAgentMetadata) bool {
    return execMD != nil && execMD.StartedAt != nil
}
```

that way we can use `if m.isAgentStarted(execMD)`, and  `if !m.isAgentStarted(execMD)` as appropriate.

Reply:
lets add read lock as well